src/assets/AssetsContractController.ts

```diff
@@ -216,6 +221,15 @@ export class AssetsContractController extends BaseController<
       throw new Error(MISSING_PROVIDER_ERROR);
     }
 
+    const { isContractAddress } = await readAddressAsContract(
+      this.ethQuery,
+      tokenAddress,
+    );
```

Comment:
I am just curious if there is any specific reason why we can't just import the `EthQuery`  in `assetsUtil `itself? So we can avoid passing `this.ethQuery`  every time the user uses `readAddressAsContract.

Reply:
Yes, the reason is because `ethQuery` is constructed with an RPC provider which changes whenever the user changes networks. `readAddressAsContract` is a non-stateful utility function so we need to pass in the ethQuery rpc wrapper since `ethQuery` needs to be re-instantiated each time the network changes.